### DISCUSSION TITLE
How to change cathode material loading in pybamm

### DISCUSSION THREAD

**Discussion by Thiyaga2025** [pybamm-team/PyBaMM#4051] [General]
hi all, sorry for this very basic question. I have some experience in battery modeling with commercial tools (GT-Autolion, BDS) but new to pybamm modeling. trying to build a model for cylindrical cell with parameters from Chen2020 dataset. I wanted to change the capacity loading(cathode), charge and discharge capacities of electrodes. how to do this and where to find this information in pybamm. 
please help me here. thanks for your time.

**Comment by TomTranter** [comment]
You can change the capacity indirectly by changing a number of other variables that go into calculating it. electrode thickness and active material volume fraction. You have to be a bit careful when changing these though as you also have to consider porosity and don't want this + active material vf summing to more than 1. Also might want to adjust nominal cell capacity so that C-rates are calculated based on the updated capacity and also cell volume if you are changing thickness. The electrode height and length also affect the total capacity of the cell and nominal capacity but not the aerial capacity. I made a little app to help adjust things consistently and also let you balance the initial lithium concentrations and combine different half cells. It might help understand what changing things has an effect on and you can download the parameter set too. https://cellcombinator.streamlit.app/

**Reply by Thiyaga2025** [reply]
hi Tom, thanks a lot for the response. sorry, I was trying to understand, hence the delay in my reply. from Chen2020 dataset, the following parameters are related to loading(g/cm^2). i.e., active material volume fraction, porosity, electrode thickness and electrode density. as you mentioned, the loading(mg/cm^2) can be modified by changing electrode thickness and active material volume fraction. so the porosity and electrode density should be calculated accordingly and hardcoded. in the app you shared, i changed positive electrode porosity and there was a change in active solid volume fraction, but electrode density and thickness remained the same. and also the summation of porosity and active material fraction is less than one. what am i missing here. Q_n and Q_p referring to overall capacity in Ampere hours? thanks.

**Reply by TomTranter** [reply]
Yes the app is slightly different to the pybamm parameters. Sorry I should have explained but there is also inactive solid which could be binder or sei or disconnected active material etc. Pybamm assumes that inactive material is 1 - porosity - active material volume fraction. I changed it slightly in the app to define active solid volume fraction where solid volume fraction is 1 - porosity. So the porosity is simply changing the fraction of solid and not solid and then the active dial is changing the fraction of active vs non active l. I find this safer in terms of conserving total volume fraction and one fewer dial to set inactive solid

**Reply by TomTranter** [reply]
You're right that density should probably also change as a volume averaged property of the electrode. Thickness affects the loading but is independent of the volume fractions

**Reply by Thiyaga2025** [reply]
Tom, thanks a ton the clarifications. this helps. will try and reach out for any help.

**Reply by Thiyaga2025** [reply]
@TomTranter , i was trying to understand the stoichiometries and got some idea from the relevant threads and "electrode state of health" pybamm worksheet. in this regard,  can you please tell me how Q_p, Q_n and Q_Li are calculated in pybamm and in your ycombinator tool. thanks.

**Reply by TomTranter** [reply]
Have a look at the ElectrodeSOH model

**Reply by Thiyaga2025** [reply]
Thanks @TomTranter ,

Q_n = parameter_values.evaluate(param.n.Q_init)
Q_p = parameter_values.evaluate(param.p.Q_init)
Q_Li = parameter_values.evaluate(param.Q_Li_particles_init)

helped to calculate.
